src/KafkaFlow.Retry/Durable/Polling/QueueTracker.cs

```diff
@@ -1,130 +1,171 @@
+            foreach (var jobDataProvider in this.jobDataProviders)
+            {
+                if (!jobDataProvider.PollingDefinition.Enabled)
                 {
-                    StdSchedulerFactory fact = new StdSchedulerFactory();
-                    fact.Initialize(new NameValueCollection { { "quartz.scheduler.instanceName", this.retryDurablePollingDefinition.Id } });
-                    this.scheduler = fact.GetScheduler(cancellationToken).GetAwaiter().GetResult();
-
-                    this.logHandler.Info(
-                        "PollingJob Scheduler Acquired",
+                    this.logHandler.Warning(
+                        "PollingJob Scheduler not enabled",
                         new
                         {
-                            PollingId = this.retryDurablePollingDefinition.Id,
-                            CronExpression = this.retryDurablePollingDefinition.CronExpression
+                            SchedulerId = this.schedulerId,
+                            PollingJobType = jobDataProvider.PollingDefinition.PollingJobType.ToString(),
+                            CronExpression = jobDataProvider.PollingDefinition.CronExpression
                         });
+
+                    continue;
                 }
 
-                if (!this.IsSchedulerActive)
-                {
-                    this.scheduler
-                        .Start(cancellationToken)
-                        .GetAwaiter()
-                        .GetResult();
+                this.ScheduleJob(jobDataProvider, cancellationToken);
+            }
+        }
 
-                    this.logHandler.Info(
-                        "PollingJob Scheduler Started",
-                        new
-                        {
-                            PollingId = this.retryDurablePollingDefinition.Id,
-                            CronExpression = this.retryDurablePollingDefinition.CronExpression
-                        });
+        internal void UnscheduleJobs(CancellationToken cancellationToken = default)
+        {
+            foreach (var jobDataProvider in this.jobDataProviders)
+            {
+                if (!jobDataProvider.PollingDefinition.Enabled)
+                {
+                    continue;
                 }
 
+                var trigger = this.triggerProvider.GetPollingTrigger(this.schedulerId, jobDataProvider.PollingDefinition);
+
+                this.logHandler.Info(
+                    "PollingJob unscheduler started",
+                    new
+                    {
+                        SchedulerId = this.schedulerId,
+                        PollingJobType = jobDataProvider.PollingDefinition.PollingJobType.ToString(),
+                        TriggerKey = trigger.Key.ToString()
+                    });
+
+                var unscheduledJob = this.scheduler
+                    .UnscheduleJob(trigger.Key)
+                    .GetAwaiter()
+                    .GetResult();
```

Comment:
Perhaps we can make this statement and the method async?

Reply:
same response from above